internal/api/v1/applications.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/fatih/color"
+	"github.com/julienschmidt/httprouter"
+	"github.com/suse/carrier/deployments"
+	"github.com/suse/carrier/internal/application"
+	"github.com/suse/carrier/paas"
+)
+
+type ApplicationsController struct {
+}
+
+func (hc ApplicationsController) Index(w http.ResponseWriter, r *http.Request) {
+	params := httprouter.ParamsFromContext(r.Context())
+	org := params.ByName("org")
+
+	client, cleanup, err := paas.NewCarrierClient(nil)
+	if handleError(w, err, 500) {
+		return
+	}
+	defer func() {
+		if cleanup != nil {
+			cleanup()
+		}
+	}()
+
+	apps, err := application.List(client.KubeClient, client.GiteaClient, org)
```

Comment:
This is a bit of a "hack". I wanted to re-use code as much as possible so I made `KubeClient`  and `GiteaClient` public in order to be able to use those here. 

After the refactoring explained in the PR description, the cli itself should not need to talk to gitea or kube. It will only talk to the carrier server. The carrier server will create its own instances of the needed clients.

Right now CarrierClient has 2 responsibilities. It's both the cli implementation (logs to stdout etc) and the thing that gathers the information from gitea, kube etc. After refactoring there will be the cli, talking only to the carrier server and presenting the results in stdout and the information gathering will be done on the server side (probably wrapped in some new structs/classes).

Reply:
Ok. Gotcha 👍🏽